src/main/java/com/apeer/sdk/impl/ApeerDevKit.java

```diff
@@ -0,0 +1,144 @@
+    private String outputParamsFile;
+
+    ApeerDevKit(ISystem system, IFileOutput fileOutputWriter) throws ApeerEnvironmentException {
+        this.system = system;
+        this.fileOutputWriter = fileOutputWriter;
+
+        log("Initializing");
+
+        var wfeInputJsonKey = "WFE_INPUT_JSON";
+        var wfe_input_json = this.system.getenv(wfeInputJsonKey);
+        if (wfe_input_json == null || wfe_input_json.isBlank()) {
+            throw new ApeerEnvironmentException("Could not find \"" + wfeInputJsonKey + "\" in environment variables");
+        }
+
+        log("Found \"" + wfeInputJsonKey + "\" to be \"" + wfe_input_json + "\"");
+
+        try {
+            inputJson = new JSONObject(wfe_input_json);
+            outputParamsFile = inputJson.getString("output_params_file");
+        } catch (JSONException ex) {
+            throw new ApeerEnvironmentException("Could not decode \"" + wfeInputJsonKey + "\"", ex);
+        }
+
+        outputJson = new JSONObject();
+
+        log("Successfully read \"" + wfeInputJsonKey + "\". Output params will be written to \"" + outputParamsFile + "\"");
+    }
+
+    /**
+     * Initializes the ADK and reads-in the WFE_INPUT_JSON. <b>NOTE</b> You must call {@code finalizeModule()} after all
+     * all other operations on this class
+     *
+     * @throws ApeerEnvironmentException When the WFE_INPUT_JSON environment variable could either
+     *                                   - not be found or
+     *                                   - it's value is not a valid JSON or
+     *                                   - it does not contain "output_params_file"
+     */
+    public ApeerDevKit() throws ApeerEnvironmentException {
+        this(new SystemFacade(), new OutputJsonFileWriter());
+    }
+
+    /**
+     * Gets the input from the WFE_INPUT_JSON environment variable
+     *
+     * @param key  The input key as defined in the module_specification.json of your module
+     * @param type The type of the input as defined in the module_specification.json of your module
+     * @param <T>  One of (all are possible as array types):
+     *             String.class, Integer.class, int.class, Double.class, double.class, Boolean.class, bool.class
+     * @return The value associated with the key as given by the WFE_INPUT_JSON environment variable
+     * @throws ApeerInputException When key could not be found or type is not supported
+     */
+    @SuppressWarnings("unchecked")
+    public <T> T getInput(String key, Class<T> type) throws ApeerInputException {
+        if (!inputJson.has(key)) {
+            throw new ApeerInputException("Could not find key \"" + key + "\" in inputs");
+        }
+
+        if (type == String.class) {
+            return (T) inputJson.getString(key);
+        } else if (type == Integer.class || type == int.class) {
+            return (T) (Integer) inputJson.getInt(key);
+        } else if (type == Double.class || type == double.class) {
```

Comment:
We should also add support for `float`. I think float is very used by imaging specialists

Reply:
JSON doesn't know the difference between float and double. Everything is double. The user can still parse this to float if he wants. I suggest we leave double to ensure max. precision